ext_test.go

```diff
@@ -44,22 +44,27 @@ func TestHang(t *testing.T) {
 	ctx1, cancel1 := context.WithTimeout(ctx, 1*time.Second)
 	defer cancel1()
 
-	peers, err := d.GetClosestPeers(ctx1, testCaseCids[0].KeyString())
-	if err != nil {
-		t.Fatal(err)
-	}
+	done := make(chan error, 1)
+	go func() {
+		_, err := d.GetClosestPeers(ctx1, testCaseCids[0].KeyString())
+		done <- err
+	}()
```

Comment:
Not really related to here, but can you add some comments into this test explaining what it is we're testing/what goes bad if the test is failing.